content/blog/microcks-realm-fix-for-apicurio.md

```diff
@@ -0,0 +1,142 @@
+
+#### Ensuring the "user" role is present correctly in the microcks-app client
+
+Once your installation is up and running, login to Keycloak with your admin credential and follow the following steps
+
+* Click on Client->*microcks-app*.
+
+* Click on Roles tab and check to confirm if the "*user*" role set correctly. If the role is not present, just create one using the "Add Role" button and give the name of the role as "*user*" and press "Save".
+
+![keycloak-microcks-app-user-role](/images/blog/keycloak-microcks-app-user-role.png)
+
+* Now, click on Client again and move to *microcks-app-js*.
+
+* Here click on "Scope" and ensure *Full Scope allowed* is "ON"
+
+![keycloak-microcks-app-js-scope](images/blog/keycloak-microcks-app-js-scope.png)
+
+
+#### Creating users in Microcks Keycloak realm
+
+Now go ahead and create a user in Microcks realm. Once a user is created, Follow the steps below to ensure that the user is setup correctly.
+
+* Ensure that the user has a role called "user" in the client role section under microcks-app client. For checking this, click on the user in the Users page and navigate to the Role Mappings tab. 
+* Here in the Client roles drop-down select *microcks-app.* You should see the an entry called "*user*" in Assigned Roles and Effective Roles sections (both should have it)
+* Once this confirmed. Navigate to Clients->*microcks-app-js*
+* Here, click on the Client scopes and then Evaluate
+* In the user input, enter the name of the user whom you want to check and then click "Evaluate"
+
+![keycloak-client-scope-evaluation.png](/home/narioinc89/Downloads/images/blog/keycloak-client-scope-evaluation.png.png)
+
+* In the form that pops up up below, click on "Effective Role Scope Mapping"
+* Here, under Client roles drop down, select *microcks-app*.
+* You should see the "*user*" role in the Granted Effective Client Role section.
+
+#### Logging in to Microcks !
+
+Now login to Microcks app and mock away !!! You should see any APIs that you posted from APICurio or specs that you manually uploaded using the Importers section, come up correctly in the *Dashboard* and the *APIs | Services* section
+
+#### Skipping TLS !!!!
+
+**Dragons ahead !! BEWARE**, at no point should this be done for production environment. TLS is one of the first steps to ensuring a strong and secure environment for the tools we are working with and at no point should you disable SSL-required configuration in Keycloak. This is ONLY for development purposes. Also this change is ONLY required when you make external calls to these services. If you provide your APICurio IP in the range of *192.168.\*.\** or *127.0.0.1* or *localhost* you can just set the SSL-requireed setting in your Keycloak realm's 'Login' setting to false (or OFF as in the UI console). 
+
+For those independent developers and coders/hobbyist, use services like Let's Encrypt to get your free TLS certs to use with these services.  

```

Comment:
Please add a markdonw link to Let's Encrypt

Reply:
Added link to Let's encrypt's website